### ISSUE TITLE
RTL (Right-to-left) Language Support

### ISSUE THREAD

**Issue by ubershmekel** [tiff/wysihtml5#70]
Arabic and Hebrew are [RTL](http://en.wikipedia.org/wiki/Right-to-left) and any rich text editor used to write in these languages needs to support it.

2 things are needed:
- An RTL/LTR button to change the http://www.w3schools.com/cssref/pr_text_direction.asp
- A few small hacks to fix strange behaviors

The best RTL support on the planet is today only seen in MSWord and other office programs. The Gmail and Google Docs rich text editors almost get it right but they only let you define RTL for each line which isn't completely correct. E.g. when you enter an English letter in the middle of a Hebrew word it shouldn't flip the entire word around.

Try adding an "a" in between these hebrew letters and see how they flip in regular text editors:

שששששללללללל

I haven't researched exactly what MSWord does so perfectly with RTL though if this feature is of interest to this project then I can help define the behavior and code as well.

**Comment by wimleers** [comment]
Solid RTL support is of course very important.

However, from the POV of a CMS (or at least _some_ CMSes), there should _not_ be a RTL/LTR button. One would set the language, and the proper text direction would be used automatically. Conclusion: this button should be configurable; not everybody wants/needs it.

RTL behavior issues: this definitely needs to be fixed.

**Comment by ubershmekel** [comment]
You're correct that some CMSes would prefer no RTL/LTR button at all. I agree that the button should be configurable.

> One would set the language, and the proper text direction would be used automatically

In my experience it's always the case that a line of English, a small acronym, an HTML bracket or markup might appear in the midst of an RTL only text. So it's OK to hide the button when LTR is the only direction. But when the default is "RTL" the button should be present. In the name of simplicity this should be a strongly recommended convention and not hard coded.

**Comment by NWSE-ICO** [comment]
Suggesting to allow the HTML dir attribute on the elements with an optional button. http://www.whatwg.org/specs/web-apps/current-work/multipage/elements.html#the-dir-attribute

(And not the CSS direction property. The "strange things" noted above are maybe related to bidirectionality http://dev.w3.org/csswg/css3-writing-modes/#text-direction , which is not completely supported in the browsers.)

By doing this, we should be able to set the direction for each inline or block element. But not for an arbitrary line that is not bound to an HTML element.

**Comment by moh6mmad** [comment]
Hey everyone
Open file wysihtml5-0.3.0.js then
in line 5514 you can use <body style="direction:rtl"></body> instead empty body
it will be OK

**Comment by alepore** [comment]
@Mohamm6d can't read what to use instead of empty body in your message.
I think that was `<body dir="rtl">`

**Comment by enesr** [comment]
You could add custom stylesheet using "stylesheets" configuration attribute:

``` javascript
$(this).wysihtml5({
    stylesheets: "../css/bootstrap-wysihtml5-rtl.css"
});
```

You can then add any RTL classes you'd need:

```
body{
    direction:rtl;
}
```

**Comment by khesayed** [comment]
- To change direction for editor elements on file bootstrap3-wysihtml5.less line 13 change float from left to right.

If you need nice wysiwyg support I18n try it https://github.com/jhollingworth/bootstrap-wysihtml5/
